### ISSUE TITLE
How to get exact r-smiles with atom-map?

### ISSUE THREAD

**Issue by Jesse-zjx** [otori-bird/retrosynthesis#8]
Thanks for your amazing work!

I am trying to get exact r-smiles with atom-map, but I am in a dilemma.
```
>>> from rdkit import Chem
>>> smi = '[c:1]1([NH:18][c:17]2[cH:16][cH:15][cH:14][c:13]([O:12][CH3:11])[cH:19]2)[n:2][c:3]([Cl:4])[n:5][c:6]2[c:7]1[nH:8][cH:9][n:10]2'
>>> mol = Chem.MolFromSmiles(smi)
>>> Chem.MolToSmiles(mol, isomericSmiles=True, rootedAtAtom=0, canonical=True)
'[c:1]1([NH:18][c:17]2[cH:16][cH:15][cH:14][c:13]([O:12][CH3:11])[cH:19]2)[n:2][c:3]([Cl:4])[n:5][c:6]2[c:7]1[nH:8][cH:9][n:10]2'
>>> [atom.ClearProp('molAtomMapNumber') for atom in mol.GetAtoms()]
[None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None]
>>> Chem.MolToSmiles(mol, isomericSmiles=True, rootedAtAtom=0, canonical=True)
'c1(Nc2cccc(OC)c2)nc(Cl)nc2nc[nH]c12'
```
As you can see, the first result comes from atom-map smiles, the second result comes from smiles without atom-map. They are different in sequence of atom(like the last atom). I wonder why and which one is the exact r-smiles? 
Should I clear atom-map before I get r-smiles?

**Comment by otori-bird** [comment]
The concept of r-smiles is applicable to chemical reactions rather than individual molecules, and discussing r-smiles for a single molecule is meaningless. To obtain r-smiles, you need to identify a specific reaction in which the molecule participates.

**Comment by Jesse-zjx** [comment]
Thank you for your answer.

Previously, I performed Chem.MolToSmiles(mol, ..., rootedAtAtom=0,...) on SMILES with atom-map and SMILES without atom-map. Different results for the same SMILES. I don’t know why atom-map affects the output of the function.

At present, I still follow the processing method of your paper. After removing the atom-map, obtain r-smiles and then map the atom-map back through the GetSubstructMatches method. Finally I got r-smiles with atom-map.

Thanks again for your research work.
